runtime/userspace/api/spdm-lib/src/codec.rs

```diff
@@ -137,6 +148,26 @@ impl<'a> MessageBuf<'a> {
         Ok(())
     }
 
+    /// Decrements the head pointer (pushes up) by specified number of bytes.
+    /// This is used to remove the headspace of the message buffer
```

Comment:
nit: I'm finding this `push` and `pull` a little confusing, since the buffer goes `head..tail`, so **decrementing** the head pointer will increase the length of the buffer?

```suggestion
    /// This is used to increase the length of the message buffer
```